### PR TITLE
Check for MEX Compiler

### PR DESCRIPTION
**Pull request by varunagrawal**
- Check if mex compiler exists for Matlab wrapper.
- Formatting and restructuring.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -103,20 +104,26 @@ if(NOT MSVC AND NOT XCODE_VERSION)
...
-# TODO: Check for matlab mex binary before handling building of binaries
-option(GTSAM_INSTALL_MATLAB_TOOLBOX      "Enable/Disable installation of matlab toolbox"  OFF)
-set(GTSAM_PYTHON_VERSION "Default" CACHE STRING "The version of Python to build the wrappers against.")
 
 # Check / set dependent variables for MATLAB wrapper
-if(GTSAM_INSTALL_MATLAB_TOOLBOX AND GTSAM_BUILD_TYPE_POSTFIXES)
-    set(CURRENT_POSTFIX ${CMAKE_${CMAKE_BUILD_TYPE_UPPER}_POSTFIX})
-endif()
+if(GTSAM_INSTALL_MATLAB_TOOLBOX)
+    find_package(Matlab COMPONENTS MEX_COMPILER REQUIRED)
```
Comment (ProfFan) [inline CMakeLists.txt:110]: I think you should probably remove the relevant code in `cmake/GtsamMatlabWrap`?

Comment (varunagrawal) [reply CMakeLists.txt:110]: I don't think there is any related code in `GtsamMatlabWrap`? I may be wrong though. My understanding is that the code in `GtsamMatlabWrap` is primarily for wrapping and I didn't want to mess with it for minor things.

**Comment by varunagrawal** [comment]
Thanks for the recommendation @ProfFan. I've reworked this PR to be a lot better.

**Comment by varunagrawal** [comment]
@ProfFan gentle reminder.

**Comment by ProfFan** [comment]
@varunagrawal Sorry I don't have capacity for this now, if you have tested on both Linux and Windows it's good I think.

**Comment by varunagrawal** [comment]
@ProfFan cool. Can you please approve this PR then? If any issues are raised for the Matlab wrapper, just assign them to me.

**Review by ProfFan — APPROVED** [review]
Sure :)

**Pull request merged**